doc/api/esm.md

```diff
@@ -1220,6 +1177,53 @@ loaded from disk but before Node.js executes it; and so on for any `.coffee`,
 `.litcoffee` or `.coffee.md` files referenced via `import` statements of any
 loaded file.
 
+#### "import map" loader
+
+The previous two loaders defined `load` hooks. This is an example of a loader
+that does its work via the `resolve` hook. This loader reads an
+`import-map.json` file that specifies which specifiers to override to another
+URL (this is a very simplistic implemenation of a small subset of the
+"import maps" specification).
+
+```js
+// import-map-loader.js
+import fs from 'node:fs/promises';
+
+const { imports } = JSON.parse(await fs.readFile('import-map.json'));
```

Comment:
```suggestion
const { imports } = JSON.parse(await fs.readFile(new URL('./import-map.json', import.meta.url)));
```

Reply:
The `import-map.json` should be given by the _user_ of the loader, so it should be taken from current working directory and not next to the source code of the loader. So I would prefer leaving it like this and not accepting your suggestion. Having said that, I don't feel strongly about it, as this is demo code, so just say the word and I'll commit the suggestion.

(obviously, cwd is naive and there should be an env variable or some such, but this is naive code for demo purpose only)

Reply:
Can you wrap it in `path.resolve` and/or add a comment? Without that, it’s a bit confusing IMO. 

> Why not just make the example the `esm.sh` one then? That’s much simpler and involves only one file, so the overall example would also be shorter.

We would have trouble with Windows compatibility, right?

Reply:
> > Why not just make the example the `esm.sh` one then? That’s much simpler and involves only one file, so the overall example would also be shorter.
> 
> We would have trouble with Windows compatibility, right?

That example I had in mind was to map all bare specifiers like `lodash` to `https://esm.sh/lodash`, and that’s it. Add a comment that `load` would fail without `--experimental-network-imports`, or without chaining the example HTTPS loader, and call it a day. It should be like a five-line example I’d think.

It’s still a bit contrived as presumably you wouldn’t always want to load the latest version of every dependency, but for an example I think it gets the idea across and then real-world implementations can fill in details.

Reply:
It seems unwise for node docs to bless an arbitrary site in the ecosystem by including it in examples.